src/components/Text.jsx

```diff
@@ -0,0 +1,23 @@
+import styled from 'styled-components';
+
+const getTextVariantStyles = (variant, theme) => {
+  const variantStyles = {
+    h1: theme.fontSizes[6],
+    h2: theme.fontSizes[5],
+    h3: theme.fontSizes[4],
+    h4: theme.fontSizes[3],
+    h5: theme.fontSizes[2],
+    h6: theme.fontSizes[1],
+    p: theme.fontSizes[2]
+  };
+
+  return variantStyles[variant] || variantStyles.h1;
+}
+
+export const Text = styled.h1({
```

Comment:
What would you do if we wanted to change the element underneath based on the variant? For example, if I want to render a `h3` element if I say the variant is `h3`

Reply:
I know about the property `as`, should I use `as` instead of `variant` or is there a way to make `variant` behave like `as`? (I'm using `variant` because MUI use this name)

Reply:
You can use the `attrs` method and combine both :smile: